src/cmd/update.js

```diff
@@ -35,7 +38,25 @@ function doUpdate(id) {
 		steps.push((next) => {
 			const binary = path.resolve(__dirname, '../../assets/updates', updates[part]);
 			const leave = partNumber === parts.length - 1;
-			dfu.write(binary, part, leave)
+
+			const parser = new ModuleParser();
+			return parser.parseFile(binary)
+				.catch(() => {
+					return;
+				})
+				.then((mod) => {
+					if (mod && mod.prefixInfo.moduleFunction !== ModuleInfo.FunctionType.BOOTLOADER) {
+						// This is a valid non-bootloader module, use FlashCommand to handle all possible quirks
+						// e.g. DROP_MODULE_INFO flag
+						return new FlashCommand().flashDfu({ binary: binary, requestLeave: leave })
+							.catch((err) => {
+								throw new Error(err);
+							});
+					} else {
+						// This is not a valid module and a simple binary, use DFU directly
+						return dfu.write(binary, part, leave);
+					}
+				})
```

Comment:
what's different about `v1.4.0` that makes this necessary?

Reply:
A mandatory SoftDevice (Nordic proprietary radio stack) upgrade.

The modules (with prefix and suffix) we generate out of the raw SoftDevice binaries provided by Nordic are somewhat special in that they contain a flag: `DROP_MODULE_INFO`. This flag indicates that before writing the module into its rightful location, the module prefix should be stripped out. It's handled by the DeviceOS when it receives a binary OTA or using YModem and it's also handled in `particle flash --usb` commands in CLI.

Reply:
got it - thanks 👍